pkg/wdk/services_chain_block_headers.go

```diff
@@ -70,13 +70,20 @@ func (c *ChainBaseBlockHeader) Hex() (string, error) {
 // followed by time, bits, and nonce fields written in little-endian order.
 // Returns an error if any of the fields cannot be parsed or written.
 func (c *ChainBaseBlockHeader) Bytes() ([]byte, error) {
-	hash, err := hex.DecodeString(c.PreviousHash)
-	if err != nil {
-		return nil, fmt.Errorf("failed to convert 'previous hash' field into bytes slice: %w", err)
-	}
+	var hash []byte
+	var err error
+
+	if c.PreviousHash == "" {
+		hash = make([]byte, 32)
+	} else {
```

Comment:
I assume the `if` is per "genesic block" - maybe, for better readability:
```suggestion
	genesis := c.PreviousHash == ""
	if genesis {
		hash = make([]byte, 32)
	} else {
```

Reply:
changed